migrations/Version202209081525261488_taoAdvancedSearch.php

```diff
@@ -25,21 +29,29 @@ public function getDescription(): string
         return 'Fix legacy configuration';
     }
 
+    /**
+     * @throws ServiceNotFoundException
+     * @throws InvalidServiceManagerException
+     * @throws common_Exception
+     */
     public function up(Schema $schema): void
     {
         /** @var SearchProxy $searchProxy */
-        $searchProxy = $this->getServiceManager()->get(SearchProxy::SERVICE_ID);
+        $searchProxy = $this->getServiceManager()->get(Search::SERVICE_ID);
 
         /** @var ConfigurableService $oldElasticSearch */
         $oldElasticSearch = $searchProxy->getAdvancedSearch();
 
         /** @var ServiceOptions $serviceOptions */
         $serviceOptions = $this->getServiceManager()->get(ServiceOptions::SERVICE_ID);
-        $serviceOptions->save(ElasticSearchConfig::class, 'hosts', $oldElasticSearch->getOption('hosts'));
+
+        if ($oldElasticSearch) {
+            $serviceOptions->save(ElasticSearchConfig::class, 'hosts', $oldElasticSearch->getOption('hosts'));
+        }
 
         $searchProxy->setOption(SearchProxy::OPTION_ADVANCED_SEARCH_CLASS, ElasticSearch::class);
 
-        $this->registerService(SearchProxy::SERVICE_ID, $searchProxy);
+        $this->registerService(Search::SERVICE_ID, $searchProxy);
```

Comment:
We should stop using this interface, is deprecated and we call SearchProxy un purpose as a first measure to delete this interface as soon as possible. 

Could you please revert it?